stdlib/2/urllib2.pyi

```diff
@@ -68,50 +82,53 @@ class HTTPErrorProcessor(BaseHandler):
     def http_response(self, request, response): ...
 
 class HTTPDefaultErrorHandler(BaseHandler):
-    def http_error_default(self, req, fp, code, msg, hdrs): ...
+    def http_error_default(self, req: Request, fp: _SupportsRead, code: int, msg: str, hdrs: Mapping[str, str]): ...
 
 class HTTPRedirectHandler(BaseHandler):
     max_repeats = ...  # type: int
     max_redirections = ...  # type: int
-    def redirect_request(self, req, fp, code, msg, headers, newurl): ...
-    def http_error_301(self, req, fp, code, msg, headers): ...
-    def http_error_302(self, req, fp, code, msg, headers): ...
-    def http_error_303(self, req, fp, code, msg, headers): ...
-    def http_error_307(self, req, fp, code, msg, headers): ...
+    def redirect_request(self, req: Request, fp: _SupportsRead, code: int, msg: str, headers: Mapping[str, str], newurl): ...
+    def http_error_301(self, req: Request, fp: _SupportsRead, code: int, msg: str, headers: Mapping[str, str]): ...
+    def http_error_302(self, req: Request, fp: _SupportsRead, code: int, msg: str, headers: Mapping[str, str]): ...
+    def http_error_303(self, req: Request, fp: _SupportsRead, code: int, msg: str, headers: Mapping[str, str]): ...
+    def http_error_307(self, req: Request, fp: _SupportsRead, code: int, msg: str, headers: Mapping[str, str]): ...
     inf_msg = ...  # type: str
 
 
 class ProxyHandler(BaseHandler):
-    def __init__(self, proxies: Optional[Any] = ...): ...
-    def proxy_open(self, req, proxy, type): ...
+    def __init__(self, proxies: Optional[Mapping[str, str]] = ...): ...
+    def proxy_open(self, req: Request, proxy, type): ...
+    proxies: Mapping[str, str]
 
 class HTTPPasswordMgr:
     def __init__(self) -> None: ...
-    def add_password(self, realm: _string, uri: Union[_string, Sequence[_string]], user: _string, passwd: _string) -> None: ...
-    def find_user_password(self, realm: _string, authuri: _string) -> Tuple[Any, Any]: ...
+    def add_password(self, realm: Optional[_string], uri: Union[_string, Sequence[_string]], user: _string, passwd: _string) -> None: ...
+    def find_user_password(self, realm: Optional[_string], authuri: _string) -> Tuple[Any, Any]: ...
```

Comment:
`Text` parameters should accept both `str` and `unicode` per PEP 484. (It's a bit hidden in the Python 2.7 section.) `unicode` won't work, since type checkers are not required to accept `str` arguments. mypy accepts both (although that might change), pytype does not.

Reply:
Fixed in f6a29d04fea7318e9bd7f94ce05c4c4afc45ae21
I only changed the type for the password handling, `_string` is still used by all the other methods.